g2p/tests/test_transducer.py

```diff
@@ -209,6 +210,42 @@ def test_deletion(self):
         self.assertEqual(self.test_deletion_transducer_csv("a").output_string, "")
         self.assertEqual(self.test_deletion_transducer_json("a").output_string, "")
 
+    def test_case_preservation(self):
+        mapping = Mapping(
+            [
+                {"in": "'a", "out": "b"},
+                {"in": "e\u0301", "out": "f"},
+                {"in": "tl", "out": "λ"},
+            ],
+            case_sensitive=False,
+            preserve_case=True,
+            norm_form="NFC",
+            case_equivalencies={"λ": "\u2144"},
+        )
+        transducer = Transducer(mapping)
+        self.assertEqual(transducer("'a").output_string, "b")
+        self.assertEqual(transducer("'A").output_string, "B")
+        self.assertEqual(transducer("E\u0301").output_string, "F")
+        self.assertEqual(transducer("E\u0301").output_string, "F")
+        # Test what happens in Heiltsuk. \u03BB should be capitalized as \u2144
+        self.assertEqual(transducer("TLaba").output_string, "\u2144aba")
+        self.assertEqual(transducer("tlaba").output_string, "λaba")
+        # I guess it's arguable what should happen here, but I'll just change case if any of the characters are differently cased
+        self.assertEqual(transducer("Tlaba").output_string, "\u2144aba")
```

Comment:
What do we do in the opposite scenario? kwk has complexities that I'd like to know are addressed. E.g., I think this is what we want:

context based:

    ᴇTO -> ETO
    ᴇto -> eto

1 char -> 2 char context based:

    ʟat -> tłat
    ʟAT -> TŁAT

2 char -> 3 char context based:

    EᴇT -> EʼA̱T
    Eᴇt -> E'a̱t

and I'll want to test what preserve_case does in all these cases when we turn it on for kwk BOAS->Umista.